src/WrappedUsdPlus.sol

```diff
@@ -12,26 +12,30 @@ import {
 } from "openzeppelin-contracts-upgradeable/contracts/token/ERC20/extensions/ERC20PermitUpgradeable.sol";
 import {Ownable2StepUpgradeable} from "openzeppelin-contracts-upgradeable/contracts/access/Ownable2StepUpgradeable.sol";
 import {UsdPlus, ITransferRestrictor} from "./UsdPlus.sol";
+import {ControlledUpgradeable} from "./deployment/ControlledUpgradeable.sol";
 
 /// @notice wrapped rebasing stablecoin
 /// @author Dinari (https://github.com/dinaricrypto/usdplus-contracts/blob/main/src/WrappedUsdPlus.sol)
-contract WrappedUsdPlus is UUPSUpgradeable, ERC4626Upgradeable, ERC20PermitUpgradeable, Ownable2StepUpgradeable {
+contract WrappedUsdPlus is UUPSUpgradeable, ERC4626Upgradeable, ERC20PermitUpgradeable, ControlledUpgradeable {
     /// ------------------ Initialization ------------------
 
     function initialize(address usdplus, address initialOwner) public initializer {
         __ERC4626_init(IERC20(usdplus));
         __ERC20Permit_init("wUSD+");
         __ERC20_init("wUSD+", "wUSD+");
-        __Ownable_init_unchained(initialOwner);
+        __AccessControlDefaultAdminRules_init_unchained(0, initialOwner);
+    }
+
+    function reinitialize(address initialOwner, address upgrader) external reinitializer(2) {
```

Comment:
`initialize` should be modified to set upgrader and version. `reinitialize` should set version.